x-pack/plugins/fleet/server/services/agent_policies/bump_agent_policies_task.ts

```diff
@@ -0,0 +1,135 @@
+            }
+
+            await runWithCache(async () => {
+              await _updatePackagePoliciesThatNeedBump(appContextService.getLogger(), cancelled);
+            });
+          },
+          async cancel() {
+            cancelled = true;
+          },
+        };
+      },
+    },
+  });
+}
+
+async function getPackagePoliciesToBump(savedObjectType: string) {
+  const result = await appContextService
+    .getInternalUserSOClientWithoutSpaceExtension()
+    .find<PackagePolicySOAttributes>({
+      type: savedObjectType,
+      filter: normalizeKuery(savedObjectType, `${savedObjectType}.bump_agent_policy_revision:true`),
+      perPage: SO_SEARCH_LIMIT,
+      namespaces: ['*'],
+      fields: ['id', 'namespaces', 'policy_ids'],
+    });
+  return {
+    total: result.total,
+    items: result.saved_objects.map((so) =>
+      mapPackagePolicySavedObjectToPackagePolicy(so, so.namespaces)
+    ),
+  };
+}
+
+export async function _updatePackagePoliciesThatNeedBump(logger: Logger, cancelled: boolean) {
+  const savedObjectType = await getPackagePolicySavedObjectType();
+  const packagePoliciesToBump = await getPackagePoliciesToBump(savedObjectType);
+
+  logger.info(
+    `Found ${packagePoliciesToBump.total} package policies that need agent policy revision bump`
+  );
+
+  const packagePoliciesIndexedBySpace = packagePoliciesToBump.items.reduce((acc, policy) => {
+    const spaceId = policy.spaceIds?.[0] ?? DEFAULT_SPACE_ID;
+    if (!acc[spaceId]) {
+      acc[spaceId] = [];
+    }
+
+    acc[spaceId].push(policy);
+
+    return acc;
+  }, {} as { [k: string]: PackagePolicy[] });
+
+  const start = Date.now();
+
+  for (const [spaceId, packagePolicies] of Object.entries(packagePoliciesIndexedBySpace)) {
+    if (cancelled) {
+      throw new Error('Task has been cancelled');
+    }
+
+    const soClient = appContextService.getInternalUserSOClientForSpaceId(spaceId);
+    const esClient = appContextService.getInternalUserESClient();
+
+    await soClient.bulkUpdate<PackagePolicySOAttributes>(
+      packagePolicies.map((item) => ({
+        type: savedObjectType,
+        id: item.id,
+        attributes: {
+          bump_agent_policy_revision: false,
+        },
+      }))
+    );
+
+    const updatedCount = packagePolicies.length;
+
+    const agentPoliciesToBump = uniq(packagePolicies.map((item) => item.policy_ids).flat());
+
+    await agentPolicyService.bumpAgentPoliciesByIds(soClient, esClient, agentPoliciesToBump);
+
+    logger.debug(
+      `Updated ${updatedCount} package policies in space ${spaceId} in ${
+        Date.now() - start
+      }ms, bump ${agentPoliciesToBump.length} agent policies`
+    );
+  }
+}
+
+export async function scheduleBumpAgentPoliciesTask(taskManagerStart: TaskManagerStartContract) {
+  await taskManagerStart.ensureScheduled({
+    id: `${TASK_TYPE}:${uuidv4()}`,
+    scope: ['fleet'],
+    params: {},
+    taskType: TASK_TYPE,
+    runAt: new Date(Date.now() + 3 * 1000),
```

Comment:
nit: how often should we run that task? it seems to me it will only needed when we have a migration so maybe running it less often like once per hour is good enough?